### PR TITLE
Improve reconnect logic

### PR DESCRIPTION
**Pull request by hiroshihorie**

### CODE REVIEW

File: Sources/LiveKit/Core/SignalClient.swift [hunk 1]
```diff
@@ -217,6 +210,9 @@ actor SignalClient: Loggable {
         await _addTrackCompleters.reset()
         await _requestQueue.clear()
         await _responseQueue.clear()
+
+        self.disconnectError = LiveKitError.from(error: disconnectError)
+        connectionState = .disconnected
```
Comment (davidzhao) [inline Sources/LiveKit/Core/SignalClient.swift:215]: why move this? does it change anything?

File: Sources/LiveKit/Participant/LocalParticipant.swift [hunk 2]
```diff
@@ -421,7 +421,7 @@ extension LocalParticipant {
             }
     }
 
-    func republishTracks() async throws {
+    func rePublishAllTracks() async throws {
```
Comment (davidzhao) [inline Sources/LiveKit/Participant/LocalParticipant.swift:424]: nit: republish does not need to be capitalized like rePublish
```suggestion
    func republishAllTracks() async throws {
```

**Review by davidzhao — COMMENTED** [review]
a few nits.. I'm not 100% about the queuing behavior.. it seems that we may potentially be queuing things when we shouldn't.

with full reconnect cases, nothing should be queued, since it's basically a new connection

File: Sources/LiveKit/Core/SignalClient.swift (hunk 1 above)
Comment (hiroshihorie) [reply Sources/LiveKit/Core/SignalClient.swift:215]: I think it changes the timing of `didUpdateConnectionState:` delegate firing.

**Review by davidzhao — APPROVED** [review]

**Pull request merged**